subsys/debug/thread_analyzer/thread_analyzer.c

```diff
@@ -170,6 +171,11 @@ static void thread_analyze_cb(const struct k_thread *cthread, void *user_data)
 	ARG_UNUSED(ret);
 
 	cb(&info);
+
+#ifdef CONFIG_THREAD_ANALYZER_LONG_FRAME_PER_INTERVAL
```

Comment:
Looks like this is already checked inside the function, so should not be necessary and will probably be optimized away anyway.

Reply:
I would prefer less `ifdef` over saving space when optimization is not enabled.
This is a nit. Leave it as is and resolve if you wish.

Reply:
From the coding style:
https://www.kernel.org/doc/html/latest/process/coding-style.html#conditional-compilation

Reply:
As I understand it `#ifdef` inside `void z_sched_thread_reset_long_frame(struct k_thread *thread)` should not be used instead the function should be marked with `__maybe_unused` so the compiler will not complain if it is unused.